superset/commands/report/execute.py

```diff
@@ -139,22 +139,37 @@ def update_report_schedule_slack_v2(self) -> None:
                     slack_recipients = json.loads(recipient.recipient_config_json)
                     # we need to ensure that existing reports can also fetch
                     # ids from private channels
+                    channels_list = get_recipients_list(slack_recipients["target"])
+                    channels_list = [channel.lstrip("#") for channel in channels_list]
+                    channels = get_channels_with_search(
+                        search_string=channels_list,
+                        types=[
+                            SlackChannelTypes.PRIVATE,
+                            SlackChannelTypes.PUBLIC,
+                        ],
+                        exact_match=True,
+                    )
+                    if len(channels_list) != len(channels):
+                        missing_channels = set(channels_list) - {
+                            channel["name"] for channel in channels
+                        }
+                        msg = (
+                            "Could not find the following channels: "
+                            f"{', '.join(missing_channels)}"
+                        )
+                        raise UpdateFailedError(msg)
```

Comment:
Do we want to block the conversion to v2 if channel ids can't be found or just log them and continue?

Reply:
@eschutho from my tests currently (before this PR) what's happening is that the execution itself is listed as successful in the execution logs, but nothing is sent to Slack. I thought about calling `send()` with the type set to `V1` again, but that would cause an infinite loop as we would validate again if there are scopes to migrate, attempt the migration, fail again, etc. Calling it with `V2` would also fail as we don't have all IDs. 

Considering that the Slack deprecation is next Tuesday,  I think it makes sense to consider any issues to migrate to V2 an error that should be fixed asap. Let me know your thoughts -- happy to think of other alternatives.

Reply:
Ok, this makes sense.

Reply:
Thanks! Won't resolve this as I think it's good context for someone looking at the changes in the future